src/components/node.js

```diff
@@ -102,6 +102,7 @@ function Node({
   onNodeSelected = () => {},
   onNodeUpdate = () => Promise.resolve(),
 }: INodeProps) {
+  const draggingEdge = useRef(false);
```

Comment:
useRef instead of useState essentially gives us an "instance variable" to replicate the class-based behavior by having event handlers use the ref instead of what would be an always-stale state variable.